nixio/pycore/h5dataset.py

```diff
@@ -43,6 +43,8 @@ def write_data(self, data, count=None, offset=None):
             self.dataset[:] = data
 
     def read_data(self, data, count=None, offset=None):
+        if 0 in self.dataset.shape or len(self.dataset.shape) == 0:
+            return
```

Comment:
I see your point, but in the worst case `shape` should be `()`, which would make `0 in shape` false (it will pass the first check with false). It's written the way it is because it started as a workaround for the weird case where the shape is `(0,)` or something like `(2, 0, 1)` (nonsensical, but valid). Both conditions (`(..., 0, ...)` and `()`) are equally rare I believe, so either order is just as valid (maybe..?).